File: client/src/api/notification.ts
import apiClient from './client'

// ==================== 타입 정의 ====================

export interface NotificationResponse {
    id: number
    type: NotificationType
    title: string
    message: string
    relatedId: number | null
    relatedType: string | null
    isRead: boolean
    createdAt: string
}

export type NotificationType =
    | 'TEAM_INVITE'
    | 'TEAM_JOIN'
    | 'TEAM_LEAVE'
    | 'SCRIM_SCHEDULED'
    | 'SCRIM_STARTED'
    | 'SCRIM_FINISHED'
    | 'MATCH_RESULT'
    | 'STRATEGY_SHARED'
    | 'COMMENT_ADDED'
    | 'GENERAL'

// ==================== API 함수 ====================

export const notificationApi = {
    /**
     * 알림 목록 조회
     */
    async getNotifications(page = 0, size = 20): Promise<NotificationResponse[]> {
        const response = await apiClient.get<NotificationResponse[]>('/notifications', {
            params: { page, size }
        })
        return response.data
    },

    /**
     * 읽지 않은 알림 조회
     */
    async getUnreadNotifications(): Promise<NotificationResponse[]> {
        const response = await apiClient.get<NotificationResponse[]>('/notifications/unread')
        return response.data
    },

    /**
     * 읽지 않은 알림 개수
     */
    async getUnreadCount(): Promise<number> {
        const response = await apiClient.get<{ count: number }>('/notifications/unread/count')
        return response.data.count
    },

    /**
     * 알림 읽음 처리
     */
    async markAsRead(id: number): Promise<boolean> {
        const response = await apiClient.patch<{ success: boolean }>(`/notifications/${id}/read`)
        return response.data.success
    },

    /**
     * 모든 알림 읽음 처리
     */
    async markAllAsRead(): Promise<number> {
        const response = await apiClient.patch<{ updated: number }>('/notifications/read-all')
        return response.data.updated
    }
}

export default notificationApi

File: server/lumia-core/src/main/kotlin/com/lumiaops/lumiacore/domain/Notification.kt
package com.lumiaops.lumiacore.domain

import com.lumiaops.lumiacore.domain.common.BaseTimeEntity
import jakarta.persistence.*
import java.time.LocalDateTime

/**
 * 알림 엔티티
 */
@Entity
@Table(name = "notifications")
class Notification(
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    val user: User,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    val type: NotificationType,

    @Column(nullable = false)
    val title: String,

    @Column(nullable = false, length = 500)
    val message: String,

    /** 관련 엔티티 ID (팀, 스크림 등) */
    val relatedId: Long? = null,

    /** 관련 엔티티 타입 */
    val relatedType: String? = null,

    /** 읽음 여부 */
    var isRead: Boolean = false,

    /** 읽은 시간 */
    var readAt: LocalDateTime? = null
) : BaseTimeEntity() {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null

    /**
     * 읽음 처리
     */
    fun markAsRead() {
        if (!isRead) {
            isRead = true
            readAt = LocalDateTime.now()
        }
    }
}

/**
 * 알림 타입
 */
enum class NotificationType {
    TEAM_INVITE,        // 팀 초대
    TEAM_JOIN,          // 팀 가입 알림 (팀장에게)
    TEAM_LEAVE,         // 팀 탈퇴 알림
    SCRIM_SCHEDULED,    // 스크림 예정
    SCRIM_STARTED,      // 스크림 시작
    SCRIM_FINISHED,     // 스크림 종료
    MATCH_RESULT,       // 매치 결과 등록
    STRATEGY_SHARED,    // 전략 공유됨
    COMMENT_ADDED,      // 코멘트 추가됨
    GENERAL             // 일반 알림
}
